bundles/org.openhab.voice.voskstt/src/main/java/org/openhab/voice/voskstt/internal/VoskSTTService.java

```diff
@@ -0,0 +1,226 @@
+import org.openhab.core.io.rest.LocaleService;
+import org.openhab.core.voice.RecognitionStartEvent;
+import org.openhab.core.voice.RecognitionStopEvent;
+import org.openhab.core.voice.STTException;
+import org.openhab.core.voice.STTListener;
+import org.openhab.core.voice.STTService;
+import org.openhab.core.voice.STTServiceHandle;
+import org.openhab.core.voice.SpeechRecognitionErrorEvent;
+import org.openhab.core.voice.SpeechRecognitionEvent;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vosk.Model;
+import org.vosk.Recognizer;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * The {@link VoskSTTService} class is a service implementation to use Vosk-API for Speech-to-Text.
+ *
+ * @author Miguel Álvarez - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = SERVICE_PID, property = Constants.SERVICE_PID + "=" + SERVICE_PID)
+@ConfigurableService(category = SERVICE_CATEGORY, label = SERVICE_NAME
+        + " Speech-to-Text", description_uri = SERVICE_CATEGORY + ":" + SERVICE_ID)
+public class VoskSTTService implements STTService {
+    private final Logger logger = LoggerFactory.getLogger(VoskSTTService.class);
+    private final ScheduledExecutorService executor = ThreadPoolManager.getScheduledPool("OH-voice-voskstt");
+    private final LocaleService localeService;
+    private VoskSTTConfiguration config = new VoskSTTConfiguration();
+    private static final String VOSK_FOLDER = Path.of(OpenHAB.getUserDataFolder(), "vosk").toString();
+    private static final String MODEL_PATH = Path.of(VOSK_FOLDER, "model").toString();
+    static {
+        Logger logger = LoggerFactory.getLogger(VoskSTTService.class);
+        File directory = new File(VOSK_FOLDER);
+        if (!directory.exists()) {
+            if (directory.mkdir()) {
+                logger.info("vosk dir created {}", VOSK_FOLDER);
+            }
+        }
+    }
+
+    @Activate
+    public VoskSTTService(@Reference LocaleService localeService) {
+        this.localeService = localeService;
+    }
+
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        this.config = new Configuration(config).as(VoskSTTConfiguration.class);
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        this.config = new Configuration(config).as(VoskSTTConfiguration.class);
+    }
+
+    @Override
+    public String getId() {
+        return SERVICE_ID;
+    }
+
+    @Override
+    public String getLabel(@Nullable Locale locale) {
+        return SERVICE_NAME;
+    }
+
+    @Override
+    public Set<Locale> getSupportedLocales() {
+        return Set.of(localeService.getLocale(null));
```

Comment:
I make that assumption because you language support depends on vosk, but you are right it makes no sense to return languages not supported by openHAB, how can I get a list of all supported locales to return here?

Reply:
The purpose here is to list the locales supported by Vosk, not openHAB. Is this list unknown? You are listing them in README.

Reply:
Then I think is better like this, as the language support depends on the availability of language models this could change without need to change the vosk version, so I think we will be adding a limitation that does not exists.
But I can change it, let me know.

Reply:
By doing that, you are probably limiting the service to one language, the default language that is setup. 
I have to check how is used this method by the dialog processor.

Reply:
But I have the feeling that the service can only work with one unique model and so one unique locale? 
Isn't it possible to find the locale from the model directory?

Reply:
Please add a comment of this kind in the method: as it is not possible to determine the language of the model that was downloaded and setup by the user, it is assumed the language of the model is matching the locale of the openHAB server.

And there is something to fix in startDialog (openHAB core): it should be checked that the provided locale is supported by each service. I will propose a fix. getSupportedLocales is currently never used.

Reply:
> And there is something to fix in startDialog (openHAB core): it should check that the provided locale is supported by each service. I will propose a fix. getSupportedLocales is currently never used.
As getSupported locales is not used right now I will suggest to just warn the user for incompatibility.

Ok I will add the comment, thanks for the help.